extensions/mqtt/processors/AbstractMQTTProcessor.cpp

```diff
@@ -20,51 +20,49 @@
 #include <utility>
 
 #include "utils/StringUtils.h"
+#include "utils/ProcessorConfigUtils.h"
 #include "core/ProcessContext.h"
 
 namespace org::apache::nifi::minifi::processors {
 
 void AbstractMQTTProcessor::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory>& /*factory*/) {
   if (auto value = context->getProperty(BrokerURI)) {
     uri_ = std::move(*value);
-    logger_->log_debug("AbstractMQTTProcessor: BrokerURI [%s]", uri_);
   }
+  logger_->log_debug("AbstractMQTTProcessor: BrokerURI [%s]", uri_);
+
+  mqtt_version_ = MqttVersions{utils::parsePropertyWithAllowableValuesOrThrow(*context, MqttVersion.getName(), MqttVersions::values())};
```

Comment:
note: there is a `parseEnumProperty` helper

Reply:
Thanks, used it!